Agents/AverageSquareMetrePriceAgent/tests/test_example_agent.py

```diff
@@ -0,0 +1,202 @@
+    print(f"Initialised successfully, created asynchronous derivation instance: {derivation_iri}")
+    
+    # Expected number of triples after derivation registration
+    triples = (cf.TBOX_TRIPLES + cf.ABOX_TRIPLES + cf.TS_TRIPLES)
+    triples += cf.TIME_TRIPLES_PER_PURE_INPUT * len(derivation_input_set) # timestamps for pure inputs
+    triples += cf.TIME_TRIPLES_PER_PURE_INPUT                             # timestamps for derivation instance
+    triples += len(derivation_input_set) + 1    # number of inputs + derivation instance type
+    triples += cf.AGENT_SERVICE_TRIPLES
+    triples += cf.DERIV_STATUS_TRIPLES
+    triples += cf.DERIV_INPUT_TRIPLES
+    triples += cf.DERIV_OUTPUT_TRIPLES
+
+    # Verify correct number of triples (incl. timestamp & agent triples)
+    assert sparql_client.getAmountOfTriples() == triples    
+
+    if local_agent_test:
+        # Patch RDB URL for TimeSeriesClient to ensure that Postgres is not accessed from inside the Container
+        # If patching an object, you have to always patch the object as used in the module, e.g. 
+        # if you have imported it in the form: from x import y in your module module, you have to patch 
+        # module.y instead of x.y. Documentation: https://docs.python.org/3/library/unittest.mock.html#id6
+        mocker.patch('avgsqmpriceagent.kg_operations.tsclient.DB_URL', rdb_url)
+        # Start the scheduler to monitor derivations if it's local agent test
+        agent._start_monitoring_derivations()
+
+    # Query timestamp of the derivation for every 20 seconds until it's updated
+    currentTimestamp_derivation = 0
+    while currentTimestamp_derivation == 0:
+        time.sleep(10)
+        currentTimestamp_derivation = cf.get_timestamp(derivation_iri, sparql_client)
+
+    # Query the output of the derivation instance
+    derivation_outputs = cf.get_derivation_outputs(derivation_iri, sparql_client)
+    print(f"Generated derivation outputs that belongsTo the derivation instance: {', '.join(derivation_outputs)}")
+    
+    # Verify that there are 2 derivation outputs (i.e. AveragePrice and Measure IRIs)
+    assert len(derivation_outputs) == 2
+    assert dm.OBE_AVERAGE_SM_PRICE in derivation_outputs
+    assert len(derivation_outputs[dm.OBE_AVERAGE_SM_PRICE]) == 1
+    assert dm.OM_MEASURE in derivation_outputs
+    assert len(derivation_outputs[dm.OM_MEASURE]) == 1
+    
+    # Verify the values of the derivation output
+    avg_iri = derivation_outputs[dm.OBE_AVERAGE_SM_PRICE][0]
+    inputs, postcode, price = cf.get_avgsqmprice_details(sparql_client, avg_iri)
+    # Verify postcode
+    assert len(postcode) == 1
+    assert postcode[0] == expected_postcode
+    # Verify price
+    assert len(price) == 1
+    assert price[0] == expected_avg
+    # Verify inputs (i.e. derived from)
+    for i in inputs:
+        for j in inputs[i]:
+            assert j in derivation_input_set
+            derivation_input_set.remove(j)
```

Comment:
This line (`derivation_input_set.remove(j)`) effectively removes the element from the list `cf.DERIVATION_INPUTS_1` as they share the same memory address, therefore `derivation_input_set` is passed in as an empty list in your third test parameterised test. This results in creating derivation with no input IRIs, hence the error you see when running multiple tests in one-go.

Maybe you want to use `copy.deepcopy(derivation_input_set)` to create a new list for each test and remove element from there.

Reply:
Updated as suggested and now all tests pass when running them 1) individuall, 2) all together via command line using `pytest -s --docker-compose=./docker-compose-test.yml` as well as 3) all together using Testing in VSCode.